Katello/katello
# Fixes #32998 - Http Proxy passwords are not making it to candlepin properly

To test:

1) Ensure that https://github.com/theforeman/foreman/pull/8412 is in your Foreman code

2) Configure a squid proxy with the following details:
```
proxy_username: pxuser
proxy: password: test]satellite@password+
```
-> This could be helpful for setting up the proxy: https://projects.theforeman.org/projects/katello/wiki/HttpProxyTesting

3) Test the proxy connection to make sure it's working

4) Set that proxy to be your default proxy

5) Go to `/subscriptions` and ensure that you do not see any errors
  -> Error example: '407 "Proxy Authentication Required"'

6) There's a bug  in Pulp 3 currently where HTTP Proxy syncing with a password doesn't work. Perhaps test that you can create a repo and check the remote in the database for the correct password

7) Also check that `::HttpProxy.default_global_content_proxy.password` in the console returns the correct password.
